### PR TITLE
Add accelerator setup guides

### PR DESCRIPTION
**Pull request by delock**
This document provide a places to hold accelerator setup guides.  It is intend to be a single place to lookup installation guides for different accelerators.  Currently CPU and XPU setup guides are added to this document and could be extended to other accelerators.

### CODE REVIEW

**Comment by delock** [comment]
Hi @tjruwase  @oelayan7 @hipudding, note we are adding a document called "Accelerator Setup Guide" hoping to put a short guide for different accelerators that users can quickly find.  Currently we have Intel CPU and XPU guide in this document.  It can be extended to other accelerators as well.

**Comment by tjruwase** [comment]
@delock, this is awesome. Please let me know when it is ready for review. Thanks!

**Comment by delock** [comment]
Hi @tjruwase this document is ready for review.  Thanks!

File: docs/_tutorials/accelerator-setup-guide.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+The Intel® oneAPI Base Toolkit (Base Kit) is a core set of tools and libraries, including an DPC++/C++ Compiler for building Deepspeed XPU kernels like fusedAdam and CPUAdam, high performance computation libraries demanded by IPEX, etc.
+For easy download, usage and more details, check [Intel oneAPI base-toolkit](https://www.intel.com/content/www/us/en/developer/tools/oneapi/base-toolkit.html).
+2. Install PyTorch, Intel extension for pytorch, Intel oneCCL Bindings for PyTorch. These packages are required in `xpu_accelerator` for torch functionality and performance, also communication backend on Intel platform. The recommended installation reference:
+https://intel.github.io/intel-extension-for-pytorch/index.html#installation?platform=gpu.
+
+3. Install DeepSpeed \
+`pip install deepspeed`
+
+## How to use DeepSpeed on Intel XPU
+DeepSpeed can launch on Intel XPU with common deepspeed command. Before that, user needs activate the oneAPI environment by: \
```
Comment (tjruwase) [inline docs/_tutorials/accelerator-setup-guide.md:119]: What does `common` deepspeed command mean here? Are you referring to the deepspeed launcher?

File: docs/_tutorials/accelerator-setup-guide.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+---
+
+# Contents
+- [Contents](#contents)
+- [Introduction](#introduction)
+- [Intel Architecture (IA) CPU](#intel-architecture-ia-cpu)
+- [Intel XPU](#intel-xpu)
+
+# Introduction
+DeepSpeed supports different accelerators from different companies.   Setup steps to run DeepSpeed on certain accelerators might be different.  This guide allows user to lookup the accelerator family they are using and setup environment for the hardware they are using.
```
Comment (tjruwase) [inline docs/_tutorials/accelerator-setup-guide.md:13]:
```suggestion
DeepSpeed supports different accelerators from different companies.   Setup steps to run DeepSpeed on certain accelerators might be different.  This guide allows user to lookup setup instructions for the accelerator family and hardware they are using.
```

File: docs/_tutorials/accelerator-setup-guide.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+4. Install DeepSpeed
+`pip install deepspeed`
+
+## How to launch DeepSpeed on Intel Architecture CPU
+DeepSpeed can launch on Intel Architecture CPU with default deepspeed command.  However, for compute intensive workloads, Intel Architecture CPU works best when each worker process runs on different set of physical CPU cores, so worker process does not compete CPU cores with each other.  To bind cores to each worker (rank), use the following command line switch for better performance.
+```
+deepspeed --bind_cores_to_rank <deepspeed-model-script>
+```
+This switch would automatically detect the number of CPU NUMA node on the host, then launch as many worker as number of NUMA nodes, and each worker to cores/memory of each NUMA node.  This ensures workers does not interfere with each other and all memory allocation is local memory which improves performance.
```
Comment (tjruwase) [inline docs/_tutorials/accelerator-setup-guide.md:44]:
```suggestion
This switch would automatically detect the number of CPU NUMA node on the host, launch the same number of workers, and bind each worker to cores/memory of a different NUMA node.  This improves performance by ensuring workers do not interfere with each other, and that all memory allocation is from local memory.
```

File: docs/_tutorials/accelerator-setup-guide.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+When user wish to get more control on the number of workers and which cores can be used by the workload, user can use the following command line switches.
```
Comment (tjruwase) [inline docs/_tutorials/accelerator-setup-guide.md:46]:
```suggestion
If a user wishes to have more control on the number of workers and specific cores that can be used by the workload, user can use the following command line switches.
```

File: docs/_tutorials/accelerator-setup-guide.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+Then set the environment variables for Intel oneCCL (assuming using conda environment).
+```
+export CPATH=${CONDA_PREFIX}/include:$CPATH
+export CCL_ROOT=${CONDA_PREFIX}
+export I_MPI_ROOT=${CONDA_PREFIX}
+export LD_LIBRARY_PATH=${CONDA_PREFIX}/lib/ccl/cpu:${CONDA_PREFIX}/lib/libfabric:${CONDA_PREFIX}/lib
+```
+
+## Optimize LLM inference with Intel Extension for PyTorch
+Intel Extension for PyTorch compatible w]th DeepSpeed AutoTP tensor parallel inference.  It allows CPU inference benefit from both DeepSpeed Automatic Tensor Parallelism and LLM optimization from Intel Extension for PyTorch.  To use Intel Extension for PyTorch, after call deepspeed.init_inference, call
```
Comment (tjruwase) [inline docs/_tutorials/accelerator-setup-guide.md:91]:
```suggestion
Intel Extension for PyTorch compatible with DeepSpeed AutoTP tensor parallel inference.  It allows CPU inference to benefit from both DeepSpeed Automatic Tensor Parallelism, and LLM optimizations of Intel Extension for PyTorch.  To use Intel Extension for PyTorch, after calling deepspeed.init_inference, call
```

**Comment by tjruwase** [comment]
@delock, I added a few comments and suggestions.

File: docs/_tutorials/accelerator-setup-guide.md (hunk 1 above)
Comment (delock) [reply docs/_tutorials/accelerator-setup-guide.md:119]: Yes, better be 'common deepspeed launching command', I'll correct this.

**Comment by delock** [comment]
> @delock, I added a few comments and suggestions.

Thanks @tjruwase for the corrections, the document looks much easy to understand now.

**Review by tjruwase — APPROVED** [review]

**Pull request merged**